### PR TITLE
[schedule 3.0] add bookmarking feature

### PR DESCRIPTION
**Pull request by malaikaaiyar**
fixes https://github.com/sparkletown/internal-sparkle-issues/issues/991

Screen Shot 2021-08-04 at 6 19 59 PM

Screen Shot 2021-08-04 at 6 20 04 PM

### CODE REVIEW

**Review by sunny-viktoryia — COMMENTED** [review]
Great! You did it so fast! 🙃 

a couple of comments:
1. Can we change the text if we toggle the "Bookmarked events" and get an empty list? Because that empty list means that the user did not save any of them, not that the schedule is empty: "No events scheduled" -> "No saved events" (text should be clarified with @sofisparkle)
2. Can we wait a little and merge the changes after https://github.com/sparkletown/sparkle/pull/1941 is merged? there is a button "download full schedule" which should become "download user schedule" (or similar) when the toggle is selected so that we could download the events that user has saved.

File: src/components/organisms/NavBarSchedule/NavBarSchedule.tsx [hunk 1]
```diff
@@ -89,6 +91,10 @@ export const NavBarSchedule: React.FC<NavBarScheduleProps> = ({
 
   const [selectedDayIndex, setSelectedDayIndex] = useState(0);
 
+  const [showPersonalisedSchedule, setShowPersonalisedSchedule] = useState(
+    false
+  );
+
```
[the remark this answers is not in the document]
Comment (malaikaaiyar) [reply src/components/organisms/NavBarSchedule/NavBarSchedule.tsx]: fixed in 93a6a946b52d88188193a56b15ea1d3d873441f7

File: src/components/organisms/NavBarSchedule/NavBarSchedule.tsx [hunk 2]
```diff
@@ -205,6 +214,18 @@ export const NavBarSchedule: React.FC<NavBarScheduleProps> = ({
...
       )} */}
+
+      <Toggler
+        name="bookmarked-toggle"
+        toggled={showPersonalisedSchedule}
+        onChange={() =>
+          showPersonalisedSchedule
+            ? setShowPersonalisedSchedule(false)
+            : setShowPersonalisedSchedule(true)
+        }
```
[the remark this answers is not in the document]
Comment (malaikaaiyar) [reply src/components/organisms/NavBarSchedule/NavBarSchedule.tsx]: fixed in 93a6a946b52d88188193a56b15ea1d3d873441f7

File: src/components/organisms/NavBarSchedule/NavBarSchedule.scss [hunk 3]
```diff
@@ -92,4 +92,9 @@ $weekday--margin-right: 10px;
       opacity: 1;
     }
   }
+
+  .Checkbox {
+    justify-content: left;
+    padding: $spacing--sm $NavBarSchedule--padding;
+  }
```
[the remark this answers is not in the document]
Comment (malaikaaiyar) [reply src/components/organisms/NavBarSchedule/NavBarSchedule.scss:99]: The `.Checkbox` class comes from the Toggler component. If I add a class to the Toggler component, the class is applied to the child input element of the `.Checkbox` label item, and as far as I know you cannot target parent elements. How should I get around this?

Screen Shot 2021-08-05 at 9 56 35 AM

**Comment by malaikaaiyar** [comment]
> Can we change the text if we toggle the "Bookmarked events" and get an empty list? Because that empty list means that the user did not save any of them, not that the schedule is empty: "No events scheduled" -> "No saved events" (text should be clarified with @sofisparkle)

I asked sofi in slack what we should do, but for now I'll make it show "No saved events"

**Comment by sofisparkle** [comment]
> Great! You did it so fast! 🙃
> 
> a couple of comments:
> 
> 1. Can we change the text if we toggle the "Bookmarked events" and get an empty list? Because that empty list means that the user did not save any of them, not that the schedule is empty: "No events scheduled" -> "No saved events" (text should be clarified with @sofisparkle)
> 2. Can we wait a little and merge the changes after [[SCHEDULE V3] Changed timeline logic + styles #1941](https://github.com/sparkletown/sparkle/pull/1941) is merged? there is a button "download full schedule" which should become "download user schedule" (or similar) when the toggle is selected so that we could download the events that user has saved.

Sounds good to me!!!

**Comment by malaikaaiyar** [comment]
> Can we change the text if we toggle the "Bookmarked events" and get an empty list? Because that empty list means that the user did not save any of them, not that the schedule is empty: "No events scheduled" -> "No saved events" (text should be clarified with @sofisparkle)

Done in 25283a6812b5c3a2e121ab775e69ac38b9503647

File: src/components/organisms/NavBarSchedule/NavBarSchedule.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (sunny-viktoryia) [reply src/components/organisms/NavBarSchedule/NavBarSchedule.scss:99]: as far as I see there is a prop `containerClassName` which adds the class name to the needed place. Does that work?

File: src/components/organisms/NavBarSchedule/NavBarSchedule.tsx [hunk 4]
```diff
@@ -205,9 +216,21 @@ export const NavBarSchedule: React.FC<NavBarScheduleProps> = ({
           </Button>
         </div>
       )} */}
+
+      <Toggler
+        name="bookmarked-toggle"
+        toggled={showPersonalisedSchedule}
+        onChange={() => setShowPersonalisedSchedule()}
```
Comment (sunny-viktoryia) [inline src/components/organisms/NavBarSchedule/NavBarSchedule.tsx:223]: will it work this way? and I think I'd rename it to `togglePersonalisedSchedule`
```suggestion
        onChange={setShowPersonalisedSchedule}
```

**Review by sunny-viktoryia — COMMENTED** [review]
Almost!!! 🙃 

just need to update the classname (see the discussion https://github.com/sparkletown/sparkle/pull/1946#discussion_r684013040)
and a small code style fix from my comment.

and then we can merge.

File: src/components/organisms/NavBarSchedule/NavBarSchedule.tsx (hunk 4 above)
Comment (malaikaaiyar) [reply src/components/organisms/NavBarSchedule/NavBarSchedule.tsx]: fixed in 2ff22f0e2bc731fcf9bbafbeeefe18dce293c065

File: src/components/organisms/NavBarSchedule/NavBarSchedule.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (malaikaaiyar) [reply src/components/organisms/NavBarSchedule/NavBarSchedule.scss]: yeah that works. fixed in 2ff22f0e2bc731fcf9bbafbeeefe18dce293c065

**Review by sunny-viktoryia — APPROVED** [review]
Looks good! 🙃

**Pull request merged**